sympy/stats/tests/test_joint_rv.py

```diff
@@ -46,6 +46,31 @@ def test_Normal():
     # assert E(X) == mu
     # assert variance(X) == sigma
 
+    # test symbolic multivariate normal densities
+    n = 3
+
+    Sg = MatrixSymbol('Sg', n, n)
+    mu = MatrixSymbol('mu', n, 1)
+    obs = MatrixSymbol('obs', n, 1)
+
+    X = MultivariateNormal('X', mu, Sg)
+    density_X = density(X)
+
+    eval_a = density_X(obs)
+    eval_b = density_X(0, 0, 0).subs({Sg: eye(3), mu: Matrix([0, 0, 0])}).doit()
+
+    assert eval_b == sqrt(2)/(4*pi**Rational(3/2))
+
+    n = symbols('n', natural=True)
+
+    Sg = MatrixSymbol('Sg', n, n)
+    mu = MatrixSymbol('mu', n, 1)
+    obs = MatrixSymbol('obs', n, 1)
+
+    X = MultivariateNormal('X', mu, Sg)
+    density_X_at_obs = density(X)(obs)
```

Comment:
It would be better to test what this is equal to like:
```
assert density_X_at_obs == ...
```

Reply:
I'm not sure how to test this. I tried adding:
```python
    expected_density = MatrixElement(
            exp(Rational(-1/2) * (obs - mu).T * Sg**-1 * (obs - mu)) *\
            det(Sg)**(-1/2) * (2*pi)**(-n/2),
        0, 0)
```
and even though the formulas are the same, `expected_density==density_X_at_obs` comes out as `False`. Trying `.subs`, `.doit`, `.simplify` all fail for `density_X_at_obs` giving one error or another... Any ideas?

Reply:
I've seen this come up a few times. I think it's something to do with the `MatrixSymbol` name being a `Symbol` rather than a `str` or perhaps the same for the `RandomVariable`.

Reply:
This is what is needed:
```
MatrixElement((exp(((S(1)/2)*mu.T - (S(1)/2)*obs.T)*Sg**(-1)*(-mu + obs))/sqrt((2*pi)**n*Determinant(Sg))), 0, 0)
```

Reply:
Amazing, thanks. Resolved.